File: python/8 kyu/385.py
# https://www.codewars.com/kata/56676e8fabd2d1ff3000000c/train/python

# Passed

def find_needle(haystack):
    result = None
    for idx, elem in enumerate(haystack):
        if elem == "needle":
            result = f"found the needle at position {idx}"
            break
    
    return result

output = find_needle(['3', '123124234', None, 'needle', 'world', 'hay', 2, '3', True, False])
print(output)

File: nodejs/8 kyu/385.js
// https://www.codewars.com/kata/5a34b80155519e1a00000009/train/javascript

// Passed

function multipleOfIndex(array) {
  let result = array.filter(
    (num, idx) => (idx === 0 && num === 0) || num % idx === 0,
  );
  return result;
}

const output = multipleOfIndex([22, -6, 32, 82, 9, 25]);
console.log(output);